generators/unifiedsearch-core/src/main/java/uk/gov/justice/services/clients/unifiedsearch/core/UnifiedSearchTransformerCache.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.justice.services.clients.unifiedsearch.core;
+
+
+import uk.gov.justice.services.clients.unifiedsearch.core.domain.UnifiedSearchDescriptor;
+import uk.gov.justice.services.clients.unifiedsearch.core.parser.UnifiedSearchDescriptorFileParserFactory;
+import uk.gov.justice.services.clients.unifiedsearch.core.parser.UnifiedSearchDescriptorParser;
+
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.apache.commons.collections.CollectionUtils;
+
+
+@ApplicationScoped
+public class UnifiedSearchTransformerCache {
+
+    @Inject
+    private UnifiedSearchFileFinder unifiedSearchFileFinder;
+
+    @Inject
+    private FileContentsAsStringLoader fileContentsAsStringLoader;
+
+    @Inject
+    private UnifiedSearchTransformerCacheUtils unifiedSearchTransformerCacheUtils;
+
+    private static final Object lock = new Object();
```

Comment:
This may not be needed.
Please see comment below about `@PostConstruct`

Reply:
please see my comment above :)

Reply:
I am aware it is going to be called once .... 
It was mainly about not having that locking complexity in place if the AppServer+CDI can do the job